iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/shareaza/ShareazaDownloadParser.java

```diff
@@ -0,0 +1,940 @@
+                        addLine(xhtml, "    			Name:                           " + btName);
+                        addLine(xhtml, "    			SHA1:                           " + btSHA1);
+                        addLine(xhtml, "    			EDONKEY:                        " + btEDONKEY);
+                        addLine(xhtml, "    			TIGER:                          " + btTIGER);
+                        addLine(xhtml, "    			MD5:                            " + btMD5);
+                    }
+
+                    int trackerIndex = read4Bytes(buffer);
+                    int trackerMode = read4Bytes(buffer);
+                    int nTrackers = read4Bytes(buffer);
+
+                    addLine(xhtml, "    TrackerIndex:          " + trackerIndex);
+                    addLine(xhtml, "    TrackerMode:           " + trackerMode);
+                    addLine(xhtml, "    Trackers:   " + nTrackers);
+
+                    for (int i = 0; i < nTrackers; i++) {
+                        addLine(xhtml, "    nTrackers[" + i + "]:   " + fileCount);
+
+                        String address = readString(buffer);
+                        int lastAccess = read4Bytes(buffer);
+                        int lastSuccess = read4Bytes(buffer);
+                        int nextTry = read4Bytes(buffer);
+                        int failures = read4Bytes(buffer);
+                        int tier = read4Bytes(buffer);
+                        int type = read4Bytes(buffer);
+
+                        addLine(xhtml, "    			Address:          " + address);
+                        addLine(xhtml, "    			LastAccess:       " + lastAccess);
+                        addLine(xhtml, "    			LastSuccess:      " + lastSuccess);
+                        addLine(xhtml, "    			NextTry:          " + nextTry);
+                        addLine(xhtml, "    			Failures:         " + failures);
+                        addLine(xhtml, "    			Tier:             " + tier);
+                        addLine(xhtml, "    			Type:             " + type);
+                    }
+
+                    int length = read4Bytes(buffer);
+                    addLine(xhtml, "    Length:                " + length);
+
+                    if (length > 0) {
+                        String nSource = readHashString(buffer, length);
+                        int infoStart = read4Bytes(buffer);
+                        int infoSize = read4Bytes(buffer);
+
+                        addLine(xhtml, "    Source:                " + nSource);
+                        addLine(xhtml, "    InfoStart:             " + infoStart);
+                        addLine(xhtml, "    InfoSize:              " + infoSize);
+                    }
+
+                    if (version >= 23) {
+                        addLine(xhtml, "TorrentSuccess:                 " + read4Bytes(buffer));
+                        addLine(xhtml, "BTH HASH:                       " + readHashString(buffer, 20));
```

Comment:
The torrentSuccess data read contains the length of data bytes that follows immediately. So, these lines should be:
                        int torrentSuccess = read4Bytes(buffer);
                        addLine(xhtml, "TorrentSuccess:                 " + torrentSuccess);
                        if (torrentSuccess >= 0) {
                            addLine(xhtml, "TorrentSuccess Data:                       " + readHashString(buffer, torrentSuccess));
                        }